java/add-enc-exchange-set/src/main/java/com/esri/arcgisruntime/sample/addencexchangeset/MainActivity.java

```diff
@@ -0,0 +1,163 @@
+/* Copyright 2019 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgisruntime.sample.addencexchangeset;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+import android.Manifest;
+import android.content.pm.PackageManager;
+import android.os.Bundle;
+import android.os.Environment;
+import android.support.annotation.NonNull;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.widget.Toast;
+
+import com.esri.arcgisruntime.geometry.Envelope;
+import com.esri.arcgisruntime.geometry.GeometryEngine;
+import com.esri.arcgisruntime.hydrography.EncCell;
+import com.esri.arcgisruntime.hydrography.EncDataset;
+import com.esri.arcgisruntime.hydrography.EncEnvironmentSettings;
+import com.esri.arcgisruntime.hydrography.EncExchangeSet;
+import com.esri.arcgisruntime.layers.EncLayer;
+import com.esri.arcgisruntime.loadable.LoadStatus;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.Viewpoint;
+import com.esri.arcgisruntime.mapping.view.MapView;
+
+public class MainActivity extends AppCompatActivity {
+
+  private static final String TAG = MainActivity.class.getSimpleName();
+
+  private MapView mMapView;
+  private Envelope mCompleteExtent;
```

Comment:
Does this need to be a member variable or can we declare it when the `EncExchangeSet` is loaded?

Reply:
It's to avoid the 'effectively final' restriction for local variables in lambdas.